clar2wasm/src/linker.rs

```diff
@@ -3603,12 +3603,12 @@ fn link_print_fn(linker: &mut Linker<ClarityWasmContext>) -> Result<(), Error> {
                     .and_then(|export| export.into_memory())
                     .ok_or(Error::Wasm(WasmError::MemoryNotFound))?;
 
-                let serialized_ty = String::from_utf8(read_bytes_from_wasm(
+                let serialized_ty = read_identifier_from_wasm(
```

Comment:
Wow, this made me realize that `read_identifier` doesn't validate that we have a correct identifier, we are just reading a string.